synapse/cores/common.py

```diff
@@ -259,67 +297,54 @@ def setModlVers(self, name, vers):
 
-        Returns:
-            (None)
-
-        Example:
+        for name, modl in modtups:
 
-            with s_cortex.openurl('ram:///') as core:
+            for tnam, tnfo in modl.get('types', ()):
+                self.addRuntNode('syn:type', tnam, **tnfo)
 
-                def v0():
-                    addModelStuff(core)
+            for fnam, fnfo, props in modl.get('forms', ()):
 
-                def v1():
-                    addMoarStuff(core)
+                self.addRuntNode('syn:form', fnam, **fnfo)
+                self.addRuntNode('syn:prop', fnam, **fnfo)
 
-                revs = [ (0,v0), (1,v1) ]
+                for pnam, pnfo in props:
+                    full = fnam + ':' + pnam
+                    self.addRuntNode('syn:prop', full, **pnfo)
 
-                core.revModlVers('foo', revs)
-
-        Each specified function is expected to update the cortex including data migration.
+    def revModlVers(self, name, vers, func):
         '''
-        if not revs:
-            return
-
-        curv = self.getModlVers(name)
+        Update and track a model version using a callback function.
 
-        maxver = revs[-1][0]
-        if maxver == curv:
-            return
+        Args:
+            name (str): The name of the model
+            vers (int): The version int ( in YYYYMMDDHHMM int format )
+            func (function):  The version update function
 
-        if not self.getConfOpt('rev:model'):
-            raise s_common.NoRevAllow(name=name, mesg='add rev:model=1 to cortex url to allow model updates')
+        Returns:
+            (None)
 
-        for vers, func in sorted(revs):
 
-            if vers <= curv:
-                continue
+        Each specified function is expected to update the cortex including data migration.
+        '''
+        if not self.getConfOpt('rev:model'):
+            raise s_common.NoRevAllow(name=name, vers=vers)
 
-            if vers and curv != -1:
-                mesg = 'Updating model [{}] from [{}] => [{}] - do *not* interrupt.'.format(name, curv, vers)
-                logger.warning(mesg)
-                self.log(logging.WARNING, mesg=mesg, name=name, curv=curv, vers=vers)
+        curv = self.getModlVers(name)
+        if curv >= vers:
+            return False
 
-            # allow the revision function to optionally return the
-            # revision he jumped to ( to allow initial override )
-            retn = func()
-            if retn is not None:
-                vers = retn
+        mesg = 'Updating model [{}] from [{}] => [{}] - do *not* interrupt.'.format(name, curv, vers)
+        logger.warning(mesg)
+        self.log(logging.WARNING, mesg=mesg, name=name, curv=curv, vers=vers)
 
-            if vers and curv != -1:
-                mesg = 'Updated model [{}] from [{}] => [{}]'.format(name, curv, vers)
-                logger.warning(mesg)
-                self.log(logging.WARNING, mesg=mesg, name=name, curv=curv, vers=vers)
+        func()
```

Comment:
Should we surround this in a try catch that raises a rev error or leave bare?

Reply:
I debated this a bit, figured letting the clean exception get through would be clearer for debugging.